### ISSUE TITLE
NullPointerException on UI thread when an empty LazyColumn is displayed

### ISSUE THREAD

**Issue by KYHSGeekCode** [JetBrains/compose-jb#1173]
Reproduce:
Click on the disabled OutlinedTextField created with the below code

OS: MacOS

Version: 
```
    kotlin("jvm") version "1.5.30"
    id("org.jetbrains.compose") version "1.0.0-alpha4-build344"
```

My code:

```kotlin
@Composable
fun platformUrlInput(viewModel: ManagePageViewModel) {
    val platformUrlEnabled by viewModel.platformUrlEnabled.collectAsState()
    val url by viewModel.meetingUrl.collectAsState()
    Row {
        Switch(checked = platformUrlEnabled, onCheckedChange = { }, enabled = false)
        Text("URL:")
        OutlinedTextField(
            shape = RoundedCornerShape(50),
            colors = TextFieldDefaults.outlinedTextFieldColors(
                focusedBorderColor = R.color.primary,
                unfocusedBorderColor = R.color.grey
            ),
            value = url,
            onValueChange = viewModel::onUrlChange,
            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Ascii),
            modifier = Modifier.fillMaxWidth(),
            enabled = platformUrlEnabled
        )
    }
}
```

Stack trace:
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at androidx.compose.ui.platform.SkiaLayer.isInLayer-k-4lQ0M(SkiaLayer.desktop.kt:128)
	at androidx.compose.ui.node.LayoutNodeWrapper.withinLayerBounds-3MmeM6k(LayoutNodeWrapper.kt:639)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:77)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.LayoutNode.hitTest-9KIMszo$ui(LayoutNode.kt:846)
	at androidx.compose.ui.node.InnerPlaceable.hitTest-9KIMszo(InnerPlaceable.kt:146)
	at androidx.compose.ui.node.LayoutNode.hitTest-9KIMszo$ui(LayoutNode.kt:846)
	at androidx.compose.ui.node.InnerPlaceable.hitTest-9KIMszo(InnerPlaceable.kt:146)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.DelegatingLayoutNodeWrapper.hitTest-9KIMszo(DelegatingLayoutNodeWrapper.kt:79)
	at androidx.compose.ui.node.LayoutNode.hitTest-9KIMszo$ui(LayoutNode.kt:846)
	at androidx.compose.ui.input.pointer.PointerInputEventProcessor.process-BIzXfog(PointerInputEventProcessor.kt:66)
	at androidx.compose.ui.input.pointer.PointerInputEventProcessor.process-BIzXfog$default(PointerInputEventProcessor.kt:50)
	at androidx.compose.ui.platform.DesktopOwner.processPointerInput-8iAsVTc$ui(DesktopOwner.desktop.kt:326)
	at androidx.compose.ui.platform.DesktopOwners.onMouseMoved(DesktopOwners.desktop.kt:193)
	at androidx.compose.ui.awt.ComposeLayer$initCanvas$3$mouseMoved$1.invoke(ComposeLayer.desktop.kt:211)
	at androidx.compose.ui.awt.ComposeLayer$initCanvas$3$mouseMoved$1.invoke(ComposeLayer.desktop.kt:210)
	at androidx.compose.ui.awt.AWTDebounceEventQueue$job$1.invokeSuspend(AWTDebounceEventQueue.desktop.kt:59)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by KYHSGeekCode** [comment]
I think `outlineCache.outline!!` is the problem.
https://github.com/androidx/androidx/blob/a1d3a68c23eb2c1e48543ee6ef4b21dccc5f3eb2/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/platform/SkiaLayer.desktop.kt#L128

**Comment by akurasov** [comment]
Could you try to see if you have the same issue on build348? Build344 behaves weird and on 348 I don't have any issues in similar code.

**Comment by KYHSGeekCode** [comment]
Hi, thank you for your suggesiton. However, I am facing the same issue on build348.
Also, I found that it was not "clicking" that caused the crash, but it crashed as soon as I moved mouse pointer while the view is being displayed.

**Comment by KYHSGeekCode** [comment]
I just found that the OutlinedTextField was not the cause. I commented it out but still crashes

**Comment by KYHSGeekCode** [comment]
I found that LazyColumn was the cause of the crash.
Maybe because the mydata.value's length is 0?
```
@Composable
fun blockListTab(viewModel: ManagePageViewModel) {
    val myData = viewModel.blockList.collectAsState()
//    val myData = listOf("Hello,", "world!", "Youtube", "Facebook", "줌", "인스타", "으악")
    LazyColumn {
        items(myData.value) { item ->
            // BlockRow(item, viewModel)
        }
    }
}
```

**Comment by KYHSGeekCode** [comment]
I am pretty sure that empty `myData.value` caused the problem!

When I changed 
```
private val _blockList = MutableStateFlow(listOf<BlockItem>())
```
to
```
private val _blockList = MutableStateFlow(listOf<BlockItem>(BlockItem(0, 0, "youtube", true)))
```
In my viewmodel,

The program did not crash anymore~

Should this phenomenon be considered as a bug or user(programmer)'s fault?

**Comment by akurasov** [comment]
could you share a whole reproducer, please?

**Comment by KYHSGeekCode** [comment]
@akurasov I created a repo for PoC!

https://github.com/KYHSGeekCode/JB-Bug-Poc/

**Comment by Sydenth** [comment]
New build 362, facing the same issue. [Minimal reproducer](https://github.com/Sydenth/Reproducer-LazyListNullPointerException)
As soon as I move my mouse into the window, the app crashes with 
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at androidx.compose.ui.platform.SkiaLayer.isInLayer-k-4lQ0M
```

This does not happen up until build 331.

**Comment by Minirogue** [comment]
I'm seeing this issue in 366.

> Should this phenomenon be considered as a bug or user(programmer)'s fault?

I'd definitely call this a bug. We should be able to have an empty `LazyColumn`.

**Comment by akurasov** [comment]
I believe workaround exists here. So shouldn't be critical.

**Comment by igordmn** [comment]
This crash should be fixed in 1.0.0-beta5. Let us know, if it is still exist on your side.

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5267">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
